speed-dreams/source-2.2.3/src/simulatedDrivingAssistance/backendUnitTests/DecisionsTests.cpp

```diff
@@ -237,3 +250,54 @@ TEST_P(DecisionTest, AccelRunIndicateTests)
 }
 INSTANTIATE_TEST_SUITE_P(AccelRunIndicateTests, DecisionTest,
                          ::testing::Values(INT_MIN, -99, -1, STANDARD_THRESHOLD_ACCEL, 0, STANDARD_THRESHOLD_ACCEL, 1, 2, 99, INT_MAX));
+
+TEST_P(DecisionTestCombinatorialFloat, MultipleIndicatorTest)
+{
+    SMediator::GetInstance()->SetInterventionType(INTERVENTION_TYPE_SHARED_CONTROL);
+
+    // Load indicators from XML used for assisting the human with visual/audio indicators.
+    char path[PATH_BUF_SIZE];
+    snprintf(path, PATH_BUF_SIZE, CONFIG_XML_DIR_FORMAT, GfDataDir());
+    IndicatorConfig::GetInstance()->LoadIndicatorData(path, SMediator::GetInstance()->GetInterventionType());
+
+    tAllowedActions allowedActions;
+    allowedActions.Steer = true;
+    allowedActions.Accelerate = true;
+
+    AccelDecision accelDecision = {};
+    accelDecision.SetInterventionAmount(std::get<0>(GetParam()));
+    accelDecision.RunInterveneCommands(allowedActions);
+    accelDecision.RunIndicateCommands();
+
+    SteerDecision steerDecision = {};
+    steerDecision.SetInterventionAmount(std::get<1>(GetParam()));
+    steerDecision.RunInterveneCommands(allowedActions);
+    steerDecision.RunIndicateCommands();
+
+    auto activeIndicators = IndicatorConfig::GetInstance()->GetActiveIndicators();
+
+    // if the accelerate amount is above the STANDARD_THRESHOLD_ACCEL, INTERVENTION_ACTION_ACCELERATE indicator should be active
+    if (accelDecision.GetInterventionAmount() > STANDARD_THRESHOLD_ACCEL || SMediator::GetInstance()->GetInterventionType() == INTERVENTION_TYPE_AUTONOMOUS_AI)
```

Comment:
Not really necessary to check the intervention type is it? We set it earlier and we know it did not change (and if it did that'd be a bug)